### PR TITLE
[AC-7441] - Fix pip/virtualenv issues in opsworks-web-python

### PR DESCRIPTION
**Pull request by philophilo**
### Changes introduced in: [AC-7441](https://masschallenge.atlassian.net/browse/AC-7441):
- limits virtualenv version
- limits buildout version number
- localizes python cookbook
- Installs python3.6, pip3, python3.6-dev

### How to test
- visit https://test4.masschallenge.org/

### farther testing
- Update `custom cookbooks` of one of the test machines
- Update stack settings to use this branch `AC-7441`
- run command `setup`

### CODE REVIEW

**Review by pbx — CHANGES_REQUESTED** [review]
Thanks for digging into this.

An important outcome is that we can create new instances for a given stack. I gave this a try, but the setup stage failed:
```
FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/lib/python3.6/dist-packages/setuptools-33.1.1-py3.6.egg'
---- End output of pip3 install setuptools==41.0.1 ----
Ran pip3 install setuptools==41.0.1 returned 1
```
I checked the stack settings, and they are using `opsworks-web-python@AC-7441` as specified in the instructions.
* Click "failure log" on the [test4b instance console](https://console.aws.amazon.com/opsworks/home?region=us-east-1#/stack/67a723c8-ea0a-49ce-a7ad-45427de81248/instances/eb4bf2dc-7e70-41f2-8cb0-46789a79319a) for the full output.

**Comment by philophilo** [comment]
@pbx I have rerun `update custom cookbooks` command and then setup. `test4b` is now running. My thought is that there is some cache on the test machine that may cause the first run to fail... Let me delete all the instances, create new ones and see if that helps without requiring multiple runs for a single machine.

**Comment by pbx** [comment]
Thanks @philophilo - we definitely want to be able to spin up a new instance and have the setup command complete successfully the first time.

File: opsworks_deploy_python/attributes/default.rb [hunk 1]
```diff
@@ -8,4 +8,4 @@
 node.default["deploy_python"]["create_dirs_before_symlink"] = ['public', 'tmp']
 node.default["deploy_python"]["packages"] = ['future']
 node.default["deploy_python"]["os_packages"] = []
-node.default["deploy_python"]["venv_options"] = '--no-site-packages --python=python3.6'
```
Comment (shanbady) [inline opsworks_deploy_python/attributes/default.rb:11]: Can we also remove the "no site packages" flag from [the impact](https://github.com/masschallenge/opsworks-web-python/blob/master/impact_deploy/recipes/default.rb#L28) deploy as well?

**Review by shanbady — CHANGES_REQUESTED** [review]

**Comment by pbx** [comment]
@philophilo I see you made some commits but since you haven't said it's ready for re-review, so I'm guessing it's not. In case you _did_ want me to re-review after those commits, check out this failure I got on setup of a new instance (setuptools related, it seems):  https://console.aws.amazon.com/opsworks/home?region=us-east-1#/stack/67a723c8-ea0a-49ce-a7ad-45427de81248/instances/a0895795-2f7e-4063-86b5-f9565220cc4d/log/8d18a76c-5be9-4167-8c86-e0b6aee6c3e1

**Comment by philophilo** [comment]
@pbx @shanbady the instances now run successfully on the first run with the changes added. In addition, thee python cookbook repo was clone to the MC github account https://github.com/masschallenge/python

**Review by shanbady — APPROVED** [review]
after the latest changes i was able to instantiate a new instance with no issues as well as re-run setup without it failing. giving this my approve.

**Review by pbx — APPROVED** [review]

**Pull request merged**